src/world/footpath.c

```diff
@@ -473,20 +474,197 @@ void footpath_provisional_update()
 	footpath_provisional_remove();
 }
 
+void sub_689726_helper(uint16 *ax, uint16 *bx, uint16 *cx, uint16 *dx, uint32 rotation)
```

Comment:
Why are these ax, bx, cx, dx we know from later on that ax is x and bx is y and cx is z. Also this function has been done before and rotation variable has not been used. I would suggest refactoring https://github.com/IntelOrca/OpenRCT2/blob/develop/src/interface/viewport.c#L104-L137 to convert it into a generic function for this use.